timesketch/api/experimental/resources.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Experimental parts of the Timesketch API."""
+
+
+from flask import jsonify
+from flask_restful import Resource
+from flask_login import login_required
+
+from timesketch.api.v1.resources import ResourceMixin
+from timesketch.lib.experimental.win_logins import win_logins
+from timesketch.lib.experimental.win_services import win_services
+from timesketch.lib.experimental.create_graph import create_graph
+from timesketch.lib.experimental.delete_graph import delete_graph
+
+
+class WinLoginsResource(Resource):
+    @login_required
+    def get(self, sketch_id):
+        return jsonify(win_logins(sketch_id))
+
+
+class WinServicesResource(Resource):
+    @login_required
+    def get(self, sketch_id):
+        return jsonify(win_services(sketch_id))
+
+
+class CreateGraphResource(ResourceMixin, Resource):
+    @login_required
+    def post(self, sketch_id):
+        result = []
+        params = {
+            u'sketch_id': sketch_id,
+            u'win_logins': win_logins(sketch_id),
+            u'win_services': win_services(sketch_id),
+        }
+        for statement in create_graph.split(';'):
+            statement = statement.strip()
+            if not statement or statement.startswith('//'):
+                continue
+            result.append(self.graph_datastore.search(
```

Comment:
This will be expensive when creating large graphs. Could we batch the statements and send in bulk?

Reply:
It doesn't matter. Elasticsearch is the bottleneck here.
On greendale dataset:
 - gathering win_logins and win_services takes about 50s
 - putting this to neo4j takes 0.14s

Reply:
Ack, makes sense.